src/duke/main/Duke.java

```diff
@@ -0,0 +1,223 @@
+            System.out.println(LINEBREAK);
+            return;
+        }catch (DukeException e){
+            System.out.println(LINEBREAK);
+            System.out.println(e.getMessage());
+            System.out.println(LINEBREAK);
+            return;
+        }
+        System.out.println(LINEBREAK);
+        if (mark){
+            System.out.println(" Nice! I've marked this task as done:");
+            System.out.println("  [X] " + list.get(index).getName());
+        }else{
+            System.out.println(" OK, I've marked this task as not done yet:");
+            System.out.println("  [ ] " + list.get(index).getName());
+        }
+        System.out.println(LINEBREAK);
+        list.get(index).setMarked(mark);
+        saveTasks();
+    }
+    public static void addNewTask(String line){
+        String taskType = line.split(" ")[0];
+        if (line.length() > taskType.length()){
+            line = line.substring(taskType.length() + 1);
+        }else{
+            line = "";
+        }
+        try {
+            switch (taskType) {
+            case "todo":
+                if (line.equals("")){
+                    throw new DukeException("Error: Argument of todo should not be empty.");
+                }
+                list.add(new Todo(line, false));
+                break;
+            case "deadline":
+                String[] taskNameAndDeadline = line.split(" /by ");
+                if (taskNameAndDeadline.length < 2){
+                    throw new DukeException("Error: A Deadline Task should have the deadline.");
+                }
+                list.add(new Deadline(taskNameAndDeadline[0], false, taskNameAndDeadline[1]));
+                break;
+            case "event":
+                String[] taskNameAndTiming = line.split(" /at ");
+                if (taskNameAndTiming.length < 2){
+                    throw new DukeException("Error: An Event Task should have the event timing.");
+                }
+                list.add(new Event(taskNameAndTiming[0], false, taskNameAndTiming[1]));
+                break;
+            default:
+                throw new DukeException("Error: Command not recognised.");
+            }
+        }catch (DukeException e){
+            System.out.println(LINEBREAK);
+            System.out.println(e.getMessage());
+            System.out.println(LINEBREAK);
+            return;
+        }
+
+        System.out.println(LINEBREAK);
+        System.out.println(" added: " + list.get(taskCount));
+        System.out.println(" Total number of tasks now: " + (taskCount + 1));
+        System.out.println(LINEBREAK);
+        taskCount++;
+        saveTasks();
+    }
+    public static void waitForInput(){
+        while (true){
+            String line = in.nextLine();
+            if (line.equals("bye")){
+                break;
+            }
+            if (line.equals("list")){
+                listTasks();
+            }else if (line.split(" ")[0].equals("mark")){
```

Comment:
Consider extracting the logic out for SLAP and to improve readability. In other words, you might want to replace `line.split(" ")[0].equals("mark")` to a Boolean attribute, say `isMark`.